packages/devextreme/js/ui/drop_down_button.js

```diff
@@ -691,14 +686,14 @@ const DropDownButton = Widget.inherit({
                 this.callBase(args);
                 break;
             case 'items':
-                this._updateDataSource(this.option('items'));
+                this._updateDataController(this.option('items'));
                 this._updateItemCollection(name);
                 break;
             case 'dataSource':
                 if(Array.isArray(value)) {
-                    this._updateDataSource(this.option('dataSource'));
+                    this._updateDataController(value);
                 } else {
-                    this._initDataSource();
+                    this._dataController.initDataSource(value);
                     this._updateKeyExpr();
```

Comment:
I think this `if-else` condition indicates some incorrect splitting of responsibility between functions.
For example if `itemsToDataSource` calls `ds.dispose` we can refactor it this way:
```javascript
if(Array.isArray(value)) {
    this._dataController.itemsToDataSource(items, this._getKey());
} else {
   this._dataController.initDataSource(value);
}
this._updateKeyExpr();
```
Now it's still a bit strange why we should call different updates on `dataSource` option change - possible it shoulf be also inside of a dataController